packages/admin/admin/src/appHeader/menuButton/AppHeaderMenuButton.tsx

```diff
@@ -9,44 +8,32 @@ export type AppHeaderMenuButtonProps = IconButtonProps;
 
 export type AppHeaderMenuButtonClassKey = IconButtonClassKey;
 
-const styles = ({ spacing }: Theme) => {
-    return createStyles<AppHeaderMenuButtonClassKey, AppHeaderMenuButtonProps>({
-        root: {
-            color: "#fff",
-            marginLeft: spacing(2),
-            marginRight: spacing(2),
-        },
-        edgeStart: {},
-        edgeEnd: {},
-        colorInherit: {},
-        colorPrimary: {},
-        colorSecondary: {},
-        disabled: {},
-        sizeSmall: {},
-        sizeMedium: {},
-        sizeLarge: {},
-        colorError: {},
-        colorInfo: {},
-        colorSuccess: {},
-        colorWarning: {},
-    });
-};
-
-function MenuButton({ children, classes, ...restProps }: AppHeaderMenuButtonProps & WithStyles<typeof styles>): React.ReactElement {
+export const AppHeaderMenuButton = (inProps: AppHeaderMenuButtonProps) => {
+    const { children: propChildren, ...restProps } = useThemeProps({ props: inProps, name: "CometAdminAppHeaderMenuButton" });
     const { toggleOpen, open } = React.useContext(MenuContext);
-    if (!children) {
-        children = open ? <HamburgerClose fontSize="large" /> : <HamburgerOpen fontSize="large" />;
-    }
+
+    const children = !propChildren ? open ? <HamburgerClose fontSize="large" /> : <HamburgerOpen fontSize="large" /> : propChildren;
 
     return (
-        <IconButton classes={classes} onClick={toggleOpen} {...restProps} size="large">
+        <Root onClick={toggleOpen} {...restProps} size="large">
```

Comment:
`restProps` should probably be moved to the end, so they can override `size`

```suggestion
        <Root onClick={toggleOpen} size="large" {...restProps}>
```